git-hooks/pre-push

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+
+# @link https://gist.github.com/vlucas/8009a5edadf8d0ff7430
+#
+# Called by "git push" after it has checked the remote status,
+# but before anything has been pushed.
+#
+# If this script exits with a non-zero status nothing will be pushed.
+#
+# Steps to install, from the root directory of your repo...
+# 1. Copy the file into your repo at `.git/hooks/pre-push`
+# 2. Set executable permissions, run `chmod +x .git/hooks/pre-push`
+# 3. Or, use `rake hooks:pre_push` to install
+#
+# Try a push to main, you should get a message `*** [Policy] Never push code directly to...`
+#
+# The commands below will not be allowed...
+# `git push origin main`
+# `git push --force origin main`
+# `git push --delete origin main`
+#
+# Set the environment variable $REALLY_PUSH_TO_MAIN to bypass this pre-push
+# hook, i.e. `REALLY_PUSH_TO_MAIN=1 git push origin main`. Note: this is only
+# intended for use in automation (i.e. CI/CD).
+
+protected_branch='main'
+
+policy="\n\n[Policy] You cannot push code directly to the "$protected_branch" branch. (Prevented with pre-push hook.)\n\t  Use --really-push-to-$protected_branch to override this policy.\n\n"
```

Comment:
This says that `--really-push-to-main` is a command-line option but it's implemented only as an env var.

Reply:
Ah great catch, thank you 👍 Yes couldn't use this flag as it gets passed to `git-push`.